components/match2/wikis/deadlock/match_summary.lua

```diff
@@ -30,6 +33,53 @@ local ICONS = {
 
 local CustomMatchSummary = {}
 
+-- Hero Ban Class
+---@class DeadlockHeroBan: MatchSummaryRowInterface
+---@operator call: DeadlockHeroBan
+---@field root Html
+---@field table Html
+local HeroBan = Class.new(
+	function(self)
+		self.root = mw.html.create('div'):addClass('brkts-popup-mapveto')
+		self.table = self.root:tag('table')
+			:addClass('wikitable-striped'):addClass('collapsible'):addClass('collapsed')
+		self:createHeader()
+	end
+)
+
+---@return self
+function HeroBan:createHeader()
+	self.table:tag('tr')
+		:tag('th'):css('width', '40%'):wikitext(''):done()
+		:tag('th'):css('width', '20%'):wikitext('Bans'):done()
+		:tag('th'):css('width', '40%'):wikitext(''):done()
+	return self
+end
+
+---@param banData {numberOfBans: integer, [1]: table, [2]: table}
+---@param gameNumber integer
+---@return self
+function HeroBan:banRow(banData, gameNumber)
+	self.table:tag('tr')
+		:tag('td'):css('float', 'left')
+			:node(CustomMatchSummary._createCharacterDisplay(banData[1], false))
+		:tag('td'):css('font-size', '80%'):node(mw.html.create('div')
+			:wikitext(Abbreviation.make(
+				'Game ' .. gameNumber,
+				'Bans in game ' .. gameNumber
+			))
+		)
+		:tag('td'):css('float', 'right')
+			:node(CustomMatchSummary._createCharacterDisplay(banData[2], true))
+	return self
+end
+
+
+---@return Html
+function HeroBan:create()
+	return self.root
+end
```

Comment:
can probably use #4878 once it is finished (possibly needs the adjustments to allow non fixed max)